torch/csrc/jit/python_compiled_function.cpp

```diff
@@ -97,15 +97,19 @@ struct CompiledFunction {
       return fn->apply(inputs);
     }
 
-    PyObject* add_trace(PyObject *args, variable_list inputs) {
+    PyObject* add_trace(PyObject *args, ParsedArgs input_info) {
       JIT_ASSERT(!is_ready_);
       // Start tracing
       AutoGradMode grad_mode(grad_enabled_);
       auto num_stages = grad_enabled_ ? fn_.nderivs_ + 1 : 1;
-      auto trace = tracer::enter(fmap<TraceInput>(inputs), num_stages);
+      auto enter_info = tracer::enter(fmap<TraceInput>(input_info.vars), num_stages);
+      auto & trace = enter_info.first;
+      variable_list new_inputs(enter_info.second.begin(),
+                               enter_info.second.end() - fn_.captured_vars_.size());
+      THPObjectPtr new_args { unflatten(new_inputs, input_info.desc) };
```

Comment:
What's going on here?

Reply:
We get new inputs, so we need to re-create the Python structure of the args, but with new Variables